utility/src/test/java/org/oppia/android/util/locale/AndroidLocaleProfileTest.kt

```diff
@@ -46,35 +51,46 @@ class AndroidLocaleProfileTest {
   /* Tests for createFrom */
 
   @Test
-  fun testCreateProfile_fromRootLocale_returnsProfileWithoutLanguageAndRegionCode() {
-    val profile = AndroidLocaleProfile.createFrom(Locale.ROOT)
+  fun testCreateProfile_fromRootLocale_returnsRootProfile() {
+    val profile = androidLocaleProfileFactory.createFrom(Locale.ROOT)
 
-    assertThat(profile.languageCode).isEmpty()
-    assertThat(profile.regionCode).isEmpty()
+    assertThat(profile).isEqualTo(RootProfile)
   }
 
   @Test
-  fun testCreateProfile_fromEnglishLocale_returnsProfileWithLanguageAndWithoutRegion() {
-    val profile = AndroidLocaleProfile.createFrom(Locale.ENGLISH)
+  fun testCreateProfile_fromEnglishLocale_returnsLanguageOnlyProfile() {
+    val profile = androidLocaleProfileFactory.createFrom(Locale.ENGLISH)
 
-    assertThat(profile.languageCode).isEqualTo("en")
-    assertThat(profile.regionCode).isEmpty()
+    val languageOnlyProfile = profile as? LanguageOnlyProfile
+    assertThat(profile).isInstanceOf(LanguageOnlyProfile::class.java)
+    assertThat(languageOnlyProfile?.languageCode).isEqualTo("en")
   }
 
   @Test
   fun testCreateProfile_fromBrazilianPortuguese_returnsProfileWithLanguageAndRegion() {
-    val profile = AndroidLocaleProfile.createFrom(brazilianPortugueseLocale)
+    val profile = androidLocaleProfileFactory.createFrom(brazilianPortugueseLocale)
 
-    assertThat(profile.languageCode).isEqualTo("pt")
-    assertThat(profile.regionCode).isEqualTo("BR")
+    val languageAndRegionProfile = profile as? LanguageAndRegionProfile
+    assertThat(profile).isInstanceOf(LanguageAndRegionProfile::class.java)
+    assertThat(languageAndRegionProfile?.languageCode).isEqualTo("pt")
+    assertThat(languageAndRegionProfile?.regionCode).isEqualTo("br")
+  }
+
+  @Test
+  fun testCreateProfile_fromKenyaLocale_returnsRegionOnlyProfile() {
```

Comment:
I suspect that chanhes in this PR also fixes the failure in https://github.com/oppia/oppia-android/issues/5046. We can now create a Profile even if we only have region code, and we nolonger use `computeIfAbsent` which caused the crash.

Reply:
Ah, that's really insightful. I'll note it in the PR description and keep an eye out for the release to see if it goes away.